src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.cs

```diff
@@ -62,6 +66,79 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                 .WithTrackingName(GenSpecTrackingName);
 
             context.RegisterSourceOutput(genSpec, ReportDiagnosticsAndEmitSource);
+
+            if (!s_hasInitializedInterceptorVersion)
+            {
+                InterceptorVersion = DetermineInterceptableVersion();
+                s_hasInitializedInterceptorVersion = true;
+            }
+        }
+
+        internal static int s_interceptorVersion;
+        internal static int InterceptorVersion
+        {
+            get
+            {
+                Debug.Assert(s_hasInitializedInterceptorVersion);
+                return s_interceptorVersion;
+            }
+            set
+            {
+                s_interceptorVersion = value;
+            }
+        }
+
+        // Used with v1 interceptor lightup approach:
+        private static bool s_hasInitializedInterceptorVersion;
+        internal static Func<SemanticModel, InvocationExpressionSyntax, CancellationToken, object>? GetInterceptableLocationFunc { get; private set; }
+        internal static MethodInfo? InterceptableLocationVersionGetDisplayLocation { get; private set; }
+        internal static MethodInfo? InterceptableLocationDataGetter { get; private set; }
+        internal static MethodInfo? InterceptableLocationVersionGetter { get; private set; }
+
+        internal static int DetermineInterceptableVersion()
+        {
+            MethodInfo? getInterceptableLocationMethod = null;
+            int? interceptableVersion = null;
+
+#if UPDATE_BASELINES
+            const string envVarName = "InterceptableAttributeVersion";
+
+#pragma warning disable RS1035 // Do not use APIs banned for analyzers
+            string? interceptableVersionString = Environment.GetEnvironmentVariable(envVarName);
+#pragma warning restore RS1035
+            if (interceptableVersionString is not null)
+            {
+                if (int.TryParse(interceptableVersionString, out int version))
+                {
+                    interceptableVersion = version;
```

Comment:
> interceptableVersion = version; [](http://example.com/codeflow?start=19&length=32)

what happen if someone set the value to any random number?

Reply:
I did add a check to throw an exception in a newer commit.